### PR TITLE
add Transformers version in pined torch repo

### PR DESCRIPTION
**Pull request by WeizhuoZhang-intel**
Test Case:

- [x] [Guilty commmit search for checking Dockerfile correctness](https://inteltf-jenk.sh.intel.com/user/weizhuo/my-views/view/weizhuoz_view/job/inductor_aws_guilty_commit_search_zwz/27/)
log: [torchbench-hf_BigBird-inference-float32-dynamic-cpp-multiple-accuracy-crash_guilty_commit.log.txt](https://github.com/chuanqi129/inductor-tools/files/14691572/torchbench-hf_BigBird-inference-float32-dynamic-cpp-multiple-accuracy-crash_guilty_commit.log.txt)

- [x] [Full scope of torch dynamo test. ](https://inteltf-jenk.sh.intel.com/user/weizhuo/my-views/view/weizhuoz_view/job/inductor_aws_regular_dashboard_release_zwz/4/)
llava model failure has been fixed by transformers update.

| suite | name | accuracy | perf | reason(reference only)
-- | -- | -- | -- | --
torchbench | torchrec_dlrm | X | X | torchrec_dlrm, AttributeError: '_OpNamespace' 'fbgemm' object has no attribute 'jagged_2d_to_dense'
torchbench | DALLE2_pytorch | X | X | DALLE2_pytorch, NotImplementedError: DALL-E 2 Not Supported on CPU
torchbench | llava | X | √ | llava, RuntimeError: Quantize only works on Float Tensor got Double
torchbench | moco | X | X | moco, NotImplementedError: DistributedDataParallel/allgather requires cuda
torchbench | simple_gpt | X | X | simple_gpt, NotImplementedError: Model requires CUDA
torchbench | timm_efficientdet | X | X | timm_efficientdet, NotImplementedError: The original model code forces the use of CUDA.
torchbench | pyhpc_equation_of_state | √ | NaN | pyhpc_equation_of_state, AssertionError: ((41616 26) (1082016 ))

### CODE REVIEW

**Comment by WeizhuoZhang-intel** [comment]
Remaining issue:
Benchmark install will uninstall transformers and re-install release version.

```
#15 [build 5/7] RUN git clone -b main https://github.com/pytorch/benchmark.git &&     cd benchmark && if [ "default" = "default" ]; then git checkout `cat /workspace/pytorch/.github/ci_commit_pins/torchbench.txt`; else git checkout default; fi && pip install --no-deps -r requirements.txt &&     pip install --no-cache Jinja2==3.1.2 markupsafe==2.0.1 beartype==0.15.0 mpmath==1.3.0 &&     python install.py --continue_on_fail &&     pip install --upgrade numpy
#15 0.257 Cloning into 'benchmark'...
#15 3.562 Note: switching to 'd6015d42d9a1834bc7595c4bd6852562fb80b30b'.
15 9.431   Attempting uninstall: transformers
#15 9.433     Found existing installation: transformers 4.38.0.dev0
#15 9.517     Uninstalling transformers-4.38.0.dev0:
#15 9.767       Successfully uninstalled transformers-4.38.0.dev0
#15 14.05   Attempting uninstall: numpy
#15 14.05     Found existing installation: numpy 1.24.3
#15 14.09     Uninstalling numpy-1.24.3:
#15 14.24       Successfully uninstalled numpy-1.24.3
#15 15.60 Successfully installed MonkeyType-23.3.0 accelerate-0.28.0 boto3-1.34.66 bs4-0.0.2 distro-1.9.0 iopath-0.1.10 numpy-1.23.5 opencv-python-4.9.0.80 patch-1.16 psutil-5.9.8 py-cpuinfo-9.0.0 pynvml-11.5.0 pytest-8.1.1 pytest-benchmark-4.0.0 submitit-1.5.1 tabulate-0.9.0 timm-0.9.7 transformers-4.38.1
```

**Comment by WeizhuoZhang-intel** [comment]
Remaining issue fixed:
image

https://inteltf-jenk.sh.intel.com/user/weizhuo/my-views/view/weizhuoz_view/job/inductor_aws_guilty_commit_search_zwz/29/artifact/2024_03_21/inductor_log/image_build.log

File: docker/Dockerfile [hunk 1]
```diff
@@ -101,12 +101,17 @@ RUN git clone -b ${TORCH_VISION_BRANCH} https://github.com/pytorch/vision.git &&
...
 ENV HUGGING_FACE_HUB_TOKEN ${HF_HUB_TOKEN}
 RUN git clone -b ${TORCH_BENCH_BRANCH} https://github.com/pytorch/benchmark.git && \
     cd benchmark && if [ "${TORCH_BENCH_COMMIT}" = "default" ]; then git checkout `cat /workspace/pytorch/.github/ci_commit_pins/torchbench.txt`; else git checkout ${TORCH_BENCH_COMMIT}; fi && pip install --no-deps -r requirements.txt && \
     pip install --no-cache Jinja2==3.1.2 markupsafe==2.0.1 beartype==0.15.0 mpmath==1.3.0 && \
-    python install.py --continue_on_fail && \
+    python install.py --continue_on_fail && cd .. && \
+    git clone -b ${TRANSFORMERS_VERSION} https://github.com/huggingface/transformers.git && \
+    cd transformers && git checkout `cat /workspace/pytorch/.ci/docker/ci_commit_pins/huggingface.txt` && git submodule sync && git submodule update --init --recursive && \
+    python setup.py bdist_wheel && \
+    pip install dist/*.whl && cd .. && \
```
Comment (chuanqi129) [inline docker/Dockerfile:114]: add `--force-reinstall` ?

Comment (WeizhuoZhang-intel) [reply docker/Dockerfile:114]: No need to --force-reinstall when ```pip install dist/*.whl```. The old version will be uninstalled.

image

**Comment by chuanqi129** [comment]
How about bisect search part? Seems we'd also add similar logic in https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/bisect_search.sh#L44 & https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/bisect_run_test.sh#L27 right?

File: docker/Dockerfile [hunk 2]
```diff
@@ -101,12 +101,17 @@ RUN git clone -b ${TORCH_VISION_BRANCH} https://github.com/pytorch/vision.git &&
     pip install dist/*.whl && cd ..
 ARG TORCH_BENCH_BRANCH=main
 ARG TORCH_BENCH_COMMIT=default
+ARG TRANSFORMERS_VERSION=main
```
Comment (chuanqi129) [inline docker/Dockerfile:104]: let's change the default value to `default`, and add if else block in below, if the value is default, we used the pined commits from pytorch, others we can try to `pip install transformers==${TRANSFORMERS_VERSION}` directly, how do you think?

**Comment by WeizhuoZhang-intel** [comment]
> How about bisect search part? Seems we'd also add similar logic in https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/bisect_search.sh#L44 & https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/bisect_run_test.sh#L27 right?

Done

File: docker/Dockerfile (hunk 2 above)
Comment (WeizhuoZhang-intel) [reply docker/Dockerfile]: Done

**Review by chuanqi129 — APPROVED** [review]

**Pull request merged**